### PR TITLE
Duty roster

### PR DESCRIPTION
**Pull request by phuihock**
This module allows easy encoding of duty roster that is usually created and
distributed in spreadsheet. It follows the common format used by
manufacturing to enroll their general workers for duty each month.

Shift codes must be defined prior to creation of duty roster.  At the moment,
shift code is exactly 1 letter. While it may seems like a non-ideal
restriction, it allows duty roster to be created very quickly (with the help
of custom widgets).

The creation of a duty roster is usually responsiblity of department
supervisor, and validated by department manager. However, this module does
not distinguish supervisor from manager. User of this module should edit
the groups accordingly.

### CODE REVIEW

File: hr_roster/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import hr_roster
```
Comment (eLBati) [inline hr_roster/__init__.py]: from . import hr_roster

Comment (phuihock) [inline hr_roster/__init__.py]: Fixed.

**Comment by dreispt** [comment]
For reference, there already has been a review process on this on [Launchpad](https://code.launchpad.net/~codekaki/openerp-hr/7.0-hr_roster/+merge/217116).

The module also implements a Department hierarchy, already done by [hr_department_sequence](https://github.com/OCA/hr/tree/7.0/hr_department_sequence), but not yet ported to 8.0.
But that is a future improvement.

:+1: to merge.

File: hr_roster/hr_roster.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    _sql_constraints = [
+        ('shift_uniq', 'unique(duty_roster_id, employee_id)',
+         'Duplicate employee entries detected for this duty roster.')]
+
+
+class hr_duty_roster(osv.Model):
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+    _name = 'hr.duty_roster'
+    _description = 'Duty Roster'
+
```
Comment (ghost) [inline hr_roster/hr_roster.py:108]: Please seperate your classes into different files.

**Comment by ghost** [comment]
The idea of this module seems great to me. I like your idea of shift codes. However, some features  seem restrictive.
First, the roster is monthly. I would rather put a date_from and date_to so that a manager can make his schedule weekly or bi-weekly.
Secondly, you should allow to leaving blank boxes. It will make it much more visual and faster to enter manually.

**Comment by phuihock** [comment]
@dufresnedavid Agree with both of your suggestions. Need a bit of time. Will implement the changes.

**Comment by charbeljc** [comment]
@phuihock Will you have some time to implement the proposed changes ? regards

**Comment by phuihock** [comment]
Hi @charbeljc and all, I'm sorry that there wasn't anything done for the last 6 months and it is seemingly abandoned. While I could still try to implement the proposed changes, I am not sure when I can deliver them. Also, 8.0/9.0 compatibility is not planned.

I will be happy to take in pull request or transfer the ownership of this module to anyone who is interested though.

**Comment by dreispt** [comment]
@phuihock Thank publishing your code you code here.
 @dufresnedavid  @charbeljc  Do you any any issues, other than possible improvements, that could prevent the module from being merged?

**Comment by ghost** [comment]
@dreispt I think this module goes against the Odoo philosophy, because of the hardcoded values in the xml view and python code. OCA modules should be flexible. If we accept this now, it will be much harder to improve it later. No improvement has been done since 2014. If this module is necessary, I guess people would have worked on it already.

**Pull request closed without merging**